### DISCUSSION TITLE
Used prisma for client and server side

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [prisma/prisma#15182] [Q&A] [answered]
I want to use Prisma to have a partially replicated database on the client side so that we can reuse the same data access requests in offline mode. The backend would be MongoDB and the client would be SQLite.

The goal, in summary, is to be able to use Prisma to request the same model on the client side as on the server side.

Does this seem possible to you with Prisma?

What would be the best practices to follow and the design errors to avoid when using Prisma?

---

_I ask this question in case, I saw that this [issue](https://github.com/prisma/prisma/issues/2443) being in progress with regard to the multiple data sources for Prisma_

**Comment by austincrim** [comment, marked as answer]
Hey @vincent-herlemont 👋,

When you refer to the "client", are you talking about running Prisma in the browser? This is not possible, but it seems like an interesting idea for offline mode.

We do have an [outstanding feature request](https://github.com/prisma/prisma/issues/4992) so feel free to upvote and leave a comment with your specific use case.

**Reply by vincent-herlemont** [reply]
@austincrim I'm talking more about using prisma with [capacitorjs](https://capacitorjs.com/) or [tauri](https://tauri.app/) in TS (event of use community project like [Brendonovich/prisma-client-rust](https://github.com/Brendonovich/prisma-client-rust) if we can code in Rust).

**Reply by austincrim** [reply]
Ah, interesting! The Prisma Client does rely on Node.js APIs, which I don't think would be available in a Capacitor or Tauri environment.

Using `prisma-client-rust` in Tauri might be possible, but I haven't seen it done. I would love to hear about your experience if you try that route.

As far as using different databases, it is possible to maintain two separate Prisma schemas that each specify a different [`datasource`](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#datasource). They could then generate the Prisma Client to [custom output locations](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#generator) and be used separately.

**Comment by 3xau1o** [comment]
you need 
* [nw.js](https://nwjs.io/) for desktop apps 
* RPCs like [qwik $server](https://qwik.dev/docs/server$/) for browser

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
